AEPTarget/Sources/Target.swift

```diff
@@ -787,26 +802,35 @@ public class Target: NSObject, Extension {
             }
             Log.debug(label: Target.LOG_TAG, "processCachedTargetRequest - Cached mbox found for \(request.name) with data \(cachedMbox.description)")
 
-            let content = extractMboxContent(mboxJson: cachedMbox) ?? request.defaultContent
-            dispatchMboxContent(event: event, content: content, responsePairId: request.responsePairId)
+            let (content, responseTokens) = extractMboxContentAndResponseTokens(mboxJson: cachedMbox)
+            let analyticsPayload = getAnalyticsForTargetPayload(json: cachedMbox)
+            let metrics = extractClickMetric(mboxJson: cachedMbox)
+
+            // package analytics payload and response tokens to be returned in request callback
+            let responsePayload = packageMboxResponsePayload(responseTokens: responseTokens, analyticsPayload: analyticsPayload, metricsAnalyticsPayload: metrics.analyticsPayload)
+
+            dispatchMboxContent(event: event, content: content ?? request.defaultContent, data: responsePayload, responsePairId: request.responsePairId)
         }
 
         return requestsToSend
     }
 
-    /// Return Mbox content from mboxJson, if any
+    /// Return Mbox content and response tokens from mboxJson, if any.
     /// - Parameters:
     ///     - mboxJson: `[String: Any]` target response dictionary
-    /// - Returns: `String` mbox content, if any otherwise returns nil
-    private func extractMboxContent(mboxJson: [String: Any]) -> String? {
+    /// - Returns: tuple containg `String` mbox content and `Dictionary` containing response tokens, if any.
+    private func extractMboxContentAndResponseTokens(mboxJson: [String: Any]) -> (content: String?, responseTokens: [String: String]?) {
```

Comment:
Can we have two separate functions for content and response tokens. That will be more cleaner.

Reply:
Not sure I understand the benefit here 🤔 You'd end up writing two functions that parse the same box dictionary to extract the content and response tokens from the same options array. I believe it's error-prone and leads to unnecessary code duplication, needless to mention multiple function invocations for reading related info.